File: frontend/src/graphql/messages/Message.ts
import MessageDTO from "./MessageDTO";

export default class Message {
    public readonly id: number;
    public text: string;
    public readonly conversationId: number;
    public readonly senderId: number;
    public readonly timeSent: Date; 


    public static fromMessageDTO(messageDTO: MessageDTO) {
        return new Message(
            messageDTO.id, 
            messageDTO.text,
            messageDTO.conversationId, 
            messageDTO.senderId, 
            new Date(messageDTO.timeSent)
        );
    }

	constructor(id: number, text: string, conversationId: number, senderId: number, timeSent: Date) {
        this.id = id;
        this.text = text;
        this.conversationId = conversationId;
        this.senderId = senderId;
        this.timeSent = timeSent; 
	}

}

File: src/main/java/com/app/Message_Backend/entities/Message.java
package com.app.Message_Backend.entities;

import com.app.Message_Backend.listeners.SaveListener;

import javax.persistence.*;
import java.util.Date;

@EntityListeners( SaveListener.class )
@Entity(name = "Messages")
public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String text;
    @Column(name = "time_sent")
    private Date timeSent;
    @ManyToOne
    @JoinColumn(name="conversation_id", nullable=false)
    private Conversation conversation;
    @Column(name = "sender_id")
    private Long senderId;

    public Message() {}

    public Message(String text, Conversation conversation, Long senderId) {
        this.text = text;
        this.conversation = conversation;
        this.senderId = senderId;
        this.timeSent = new Date();
    }

    public Message(String text, Long senderId) {
        this.text = text;
        this.senderId = senderId;
        this.timeSent = new Date();
    }

    public Message(String text, Conversation conversation, Long senderId, Date timeSent) {
        this.text = text;
        this.conversation = conversation;
        this.senderId = senderId;
        this.timeSent = timeSent;
    }


    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public Long getId() {
        return id;
    }

    public Date getTimeSent() {
        return timeSent;
    }

    public void setTimeSent(Date timeSent) {
        this.timeSent = timeSent;
    }

    public Conversation getConversation() {
        return conversation;
    }

    public void setConversation(Conversation conversation) {
        this.conversation = conversation;
    }

    public Long getSenderId() {
        return senderId;
    }

    public void setSenderId(Long senderId) {
        this.senderId = senderId;
    }

    public Long getConversationId() {
        return this.conversation.getId();
    }

}
